### DISCUSSION TITLE
👑 [需求] pro组件添加有类似ztree  的组件

### DISCUSSION THREAD

**Discussion by goodve** [ant-design/pro-components#433] [Ideas]
### 🥰 需求描述

目前antd的tree 在后台管理系统中使用起来不是很方便
![image](https://user-images.githubusercontent.com/17634299/92566245-7cc8ff00-f2ae-11ea-9a9f-ffc0eedf7eb4.png)

**Comment by chenshuai2144** [comment]
听起来需要一个 ProTree

**Comment by goodve** [comment]
> 听起来需要一个 ProTree

mit 协议的  rc-tree  定制一套  antd皮肤直接上？ 🥰

**Comment by chenshuai2144** [comment]
我看了一下这个，不如antd 的 tree 强大，还不支持虚拟滚动

你喜欢什么功能？

**Comment by goodve** [comment]
> 我看了一下这个，不如antd 的 tree 强大，还不支持虚拟滚动
> 
> 你喜欢什么功能？

勾选和取消时的  父子节点是否联动  
![image](https://user-images.githubusercontent.com/17634299/92578650-b9045b80-f2be-11ea-948e-364072578ed5.png)

还有各种事件 有before的方法
![image](https://user-images.githubusercontent.com/17634299/92578852-084a8c00-f2bf-11ea-94db-df3c94c2a26b.png)

**Comment by chenshuai2144** [comment]
这个 antd 也是有的。

**Comment by itsuki0927** [comment]
@chenshuai2144 期待出现这个组件, 我尝试着去仿照`ProFormSelect`写一个`ProFormTreeSelect`,奈何总感觉怪里怪气的, 感觉不是很不统一.

**Comment by chenshuai2144** [comment]
暂时没有时间的 你可以把碰到的问题写在这里我们做的时候就可以参考了

**Comment by kaoding** [comment]
https://github.com/ant-design/pro-components/issues/259#issuecomment-677996804

**Comment by itsuki0927** [comment]
1. 不清楚`ProFormTreeSelect`要支持或者扩展哪一些 API (因为个人没有参与到ProField、ProForm 的基础组件中).
2. `valueEnum`用来统一 table，descriptions, 如果要支持 TreeSelect, valueEnum 类型只能原有基础上扩展`OriginalType & TreeSelectProps `,并且现在是如果传入了`valueEnum`默认渲染出来`ProFormSelect`,如果继续复用`valueEnum`的话,则需要再传入`valueType`,以便能够区分`Select`还是`TreeSelect`这里就需要扩展`ValueType`的类型, 或者说不使用`valueEnum`,这里怎么做选择.

**Comment by itsuki0927** [comment]
@kaoding 这是ProTree的需求点嘛?

**Comment by kaoding** [comment]
> @kaoding 这是ProTree的需求点嘛?
这个不是官方的需求，现在是自己在项目过程做出来的一个树组件，看是否有需求提为开源的公用组件，针求点意见后，再提PR。
